src/main/java/com/uid2/admin/secret/IEncryptionKeyManager.java

```diff
@@ -4,4 +4,9 @@
 
 public interface IEncryptionKeyManager {
     EncryptionKey addSiteKey(int siteId) throws Exception;
+
+    /**
+     * Creates a site key, if none exists. If created, the key is active immediately.
+     */
+    EncryptionKey ensureSiteKeyExists(int siteId) throws Exception;
```

Comment:
I've seen "ensure" used to mean "check or throw"
Maybe something like "getOrCreateSiteKey"?

Reply:
👍 Renamed to `createSiteKeyIfNoneExists`